riscv/triggers.h

```diff
@@ -29,6 +36,21 @@ typedef enum {
   TIMING_AFTER = 1
 } timing_t;
 
+typedef enum {
+  SSELECT_IGNORE = 0,
+  SSELECT_SCONTEXT = 1,
+  SSELECT_ASID = 2,
+  SSELECT_MAXVAL = 2
+} sselect_t;
+
+typedef enum {
+  MHSELECT_IGNORE = 0,
+  MHSELECT_MCONTEXT = 4,
+  MHSELECT_MCONTEXT_SELECT = 1 /* and 5 */,
+  MHSELECT_VMID_SELECT = 2 /* and 6 */,
+  MHSELECT_RESERVED = 3 /* and 7 */
```

Comment:
This is pretty ugly. I guess this is why you can't use type `mhselect_t` for the `mhselect` attribute. Is there any reason not to have the enum cover all possible cases? There are assumptions within `legalize_mhselect()` about the values of these enums, making it messy.

Reply:
Change the ```mhselect_t``` to ```mhselect_mode_t``` and add a function maps the variable ```mhselect``` to ```mhselect_mode_t```.